libkbfs/reporter_kbpki.go

```diff
@@ -235,15 +235,17 @@ func (r *ReporterKBPKI) NotifySyncStatus(ctx context.Context,
 // SuppressNotifications implements the Reporter interface for ReporterKBPKI.
 func (r *ReporterKBPKI) SuppressNotifications(
 	ctx context.Context, suppressDuration time.Duration) {
-	r.suppressCh <- suppressDuration
+	select {
+	case r.suppressCh <- suppressDuration:
+	case <-ctx.Done():
```

Comment:
This is the incoming call's ctx, but it should still eventually timeout if something is waiting on it during shutdown.

Reply:
You could save the canceler context in the reporter, and select on both contexts.

Reply:
I think it'd still be racy. The channels can be closed at the same time sending to `r.suppressCh` happens, and it's a random selection here.

Reply:
if `suppressCh` successfully receives the `suppressDuration`, then it'll get processed by the loop.

Reply:
Oh you are saying we just select on that context, but still keep suppressCh open in shutdown?

Reply:
yep